src/core/server/metrics/integration_tests/server_collector.test.ts

```diff
@@ -195,6 +195,9 @@ describe('ServerMetricsCollector', () => {
 
     waitSubject.next('go');
     await Promise.all([res1, res2]);
+    // Give the event-loop one more cycle to allow concurrent connections to be
```

Comment:
This is backported from the Node 16 upgrade - https://github.com/elastic/kibana/blame/main/src/core/server/metrics/integration_tests/server_collector.test.ts#L199

Reply:
Is there is a reason to not use `timers/promises` like that RP or `process.nextTick()`?

Reply:
Any reason not to use `process.nextTick()`?

Reply:
It resolves earlier than the connection count update.  I can dig more but the call stack is fairly deep,  preferences on timing?

Reply:
Probably fine to keep consistent with `main`